src/Analyzers/CSharp/CodeFixes/RemoveRedundantElseStatement/RemoveRedundantElseStatementCodeFixProvider.cs

```diff
@@ -0,0 +1,110 @@
+        public RemoveRedundantElseStatementCodeFixProvider()
+        {
+        }
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(IDEDiagnosticIds.RemoveRedundantElseStatementDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            RegisterCodeFix(context, CSharpAnalyzersResources.Remove_redundant_else_statement, nameof(CSharpAnalyzersResources.Remove_redundant_else_statement));
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                var elseClause = (ElseClauseSyntax)diagnostic.AdditionalLocations[0].FindNode(cancellationToken);
+                var topMostIf = FindTopMostIf(elseClause);
+                var block = topMostIf.Parent as BlockSyntax;
+
+                var root = editor.OriginalRoot;
+                var updatedRoot = root.ReplaceNode(block, RewriteBlock(block, topMostIf, elseClause));
+
+                editor.ReplaceNode(root, updatedRoot);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private static IfStatementSyntax FindTopMostIf(ElseClauseSyntax elseClause)
+        {
+            SyntaxNode node = elseClause;
+
+            while (node.Parent is IfStatementSyntax or ElseClauseSyntax)
+            {
+                node = node.Parent;
+            }
+
+            return node as IfStatementSyntax;
+        }
+
+        private static BlockSyntax RewriteBlock(BlockSyntax block, IfStatementSyntax topMostIf, ElseClauseSyntax elseClause)
+        {
+            var topMostIfIdx = block.Statements.IndexOf(topMostIf);
+
+            var newTopMostIf = topMostIf.RemoveNode(elseClause, SyntaxRemoveOptions.KeepEndOfLine);
+            var newBlock = block.ReplaceNode(topMostIf, newTopMostIf);
+            var newStatements = newBlock.Statements.InsertRange(topMostIfIdx + 1, Expand(elseClause));
+
+            return newBlock.WithStatements(newStatements);
+        }
+
+        private static ImmutableArray<StatementSyntax> Expand(ElseClauseSyntax elseClause)
+        {
+            using var _ = ArrayBuilder<StatementSyntax>.GetInstance(out var result);
+
+            switch (elseClause.Statement)
+            {
+                case BlockSyntax block:
+                    result.AddRange(block.Statements);
+                    break;
+
+                case StatementSyntax anythingElse:
+                    result.Add(anythingElse);
+                    break;
+            }
+
+            return result
+                .Select(statement => statement.WithAdditionalAnnotations(Formatter.Annotation))
+                .AsImmutable();
```

Comment:
instead of this, it would be better to just dedent the statements.  We have some patterns for that.  you can see it in action here: https://github.com/dotnet/roslyn/blob/20f6ef9569054b497983b369e7c4ca651ac4b798/src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs#L119-L157

Reply:
this seems awfully complicated and I'm not sure how to adapt it to work on a single `StatementSyntax`, why do you think it is better than just simply formatting everything inside else block?

Reply:
i'm happy to do the indentation/dedentatino part of this PR if that would help :)

Reply:
if it isn't too much trouble for you that would be very appreciated!

Reply:
Yup yup. I can look this week. Please ping me Mon/tues

Reply:
ping! 😄

Reply:
Will work on this tomorrow!

Reply:
> this seems awfully complicated

Currently it's complicated, but it produces a tremendously better result for users who don't use the formatter regularly so hopefully over time it becomes easier/standard.